services/QuillLMS/app/models/auth_credential.rb

```diff
@@ -41,6 +45,6 @@ def refresh_token_valid?
   end
 
   def refresh_token_expires_at
-    expires_at + EXPIRATION_DURATION
+    expires_at + GOOGLE_EXPIRATION_DURATION
```

Comment:
This might be overkill, but since this generic code is now serving multiple providers, it might be worth defensively coding these methods to only return values for `google` auth credentials to prevent future edge cases, e.g.
```ruby
def refresh_token_expires_at
  return nil unless google?
  
  expires_at + GOOGLE_EXPIRATION_DURATION
end

def refresh_token_valid?
 return nil unless google?
 ...
end

Reply:
This is a good point.